File: benchmarks/tsb/bench_str_title.ts
/**
 * Benchmark: str_title — str.title() titlecase conversion on 100k strings.
 * Outputs JSON: {"function": "str_title", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Array.from({ length: ROWS }, (_, i) => `hello world example ${i % 500} foo bar`);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  s.str.title();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.str.title();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "str_title",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_str_title.py
"""
Benchmark: str_title — pandas Series.str.title() titlecase conversion on 100k strings.
Outputs JSON: {"function": "str_title", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import time

import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = [f"hello world example {i % 500} foo bar" for i in range(ROWS)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.str.title()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.str.title()
total = (time.perf_counter() - start) * 1000  # ms

print(
    json.dumps(
        {
            "function": "str_title",
            "mean_ms": total / ITERATIONS,
            "iterations": ITERATIONS,
            "total_ms": total,
        }
    )
)
